custom/icds_reports/static/js/directives/mobile-filters/mobile-filters.directive.js

```diff
@@ -15,6 +15,8 @@ function MobileFiltersController($scope) {
     $scope.applyFilters = function () {
         $scope.hasLocation = false;
         $scope.hasDate = false;
+        $scope.receivedAdditionalFilterData = !($scope.$ctrl.filters.indexOf('gender') === -1 ||
+            $scope.$ctrl.filters.indexOf('age') === -1 );
```

Comment:
I find parsing this logic quite difficult, possibly because I don't quite understand the variable name and also the conditional is a little odd. I think it's: `!(genderFilterExists || ageFilterExists)`, yeah? So if either of them exist then `receivedAdditionalFilterData` `false`? Can you explain why that would be the right logic (possibly with a comment?).

Reply:
Sure I will add code comment